File: Kotlin/Demo/src/main/kotlin/Event.kt

import java.time.LocalDateTime

class Event {
    var datetime: LocalDateTime? = null
        get() = field
        set(value) {
            field = value
        }

    var datetimeEpoch: Long? = null
        get() = field
        set(value) {
            field = value
        }

    var type: String? = null
        get() = field
        set(value) {
            field = value
        }

    var latitude: Double? = null
        get() = field
        set(value) {
            field = value
        }

    var longitude: Double? = null
        get() = field
        set(value) {
            field = value
        }

    var distance: Double? = null
        get() = field
        set(value) {
            field = value
        }

    var description: String? = null
        get() = field
        set(value) {
            field = value
        }

    var size: Double? = null
        get() = field
        set(value) {
            field = value
        }
}

File: C++/Demo/Event.cpp
#include "Event.h"

tm Event::getDatetime() {
	return datetime;
}

void Event::setDatetime(tm t) {
	datetime = t;
}

long Event::getDatetimeEpoch() {
	return datetimeEpoch;
}

void Event::setDatetimeEpoch(long d) {
	datetimeEpoch = d;
}

string Event::getType() {
	return type;
}

void Event::setType(string t) {
	type = t;
}

double Event::getLatitude() {
	return latitude;
}

void Event::setLatitude(double l) {
	latitude = l;
}

double Event::getLongitude() {
	return longitude;
}

void Event::setLongitude(double l) {
	longitude = l;
}

double Event::getDistance() {
	return distance;
}

void Event::setDistance(double d) {
	distance = d;
}

string Event::getDescription() {
	return description;
}

void Event::setDescription(string d) {
	description = d;
}

double Event::getSize() {
	return size;
}

void Event::setSize(double s) {
	size = s;
}